routers/api/packages/container/container.go

```diff
@@ -148,16 +150,25 @@ func DetermineSupport(ctx *context.Context) {
 // If the current user is anonymous, the ghost user is used unless RequireSignInView is enabled.
 func Authenticate(ctx *context.Context) {
 	u := ctx.Doer
+	packageScope := auth_service.GetAccessScope(ctx.Data)
 	if u == nil {
 		if setting.Service.RequireSignInView {
 			apiUnauthorizedError(ctx)
 			return
 		}
 
 		u = user_model.NewGhostUser()
+	} else {
+		has1, _ := packageScope.HasScope(auth_model.AccessTokenScopeReadPackage)
+		has2, _ := packageScope.HasScope(auth_model.AccessTokenScopeWritePackage)
+		has3, _ := packageScope.HasScope(auth_model.AccessTokenScopeAll)
```

Comment:
![image](https://github.com/user-attachments/assets/1c67631a-414b-4283-9b8f-cb41f99a1de3)
hasscope support multiple input, why do this three times?

Reply:
`HasScope` means having all these scopes. But here, we need to check whether none of these permissions have been contained.

Reply:
I see, maybe a new function is better.

Reply:
57949b0a719e29fb8a7c261887d8f9eb269937a0